tests/unit/test_step.py

```diff
@@ -175,3 +178,136 @@ def test_composite_step(default_config_params) -> None:
             edge_attrs["output_slot"].name
             == expected_edges[(source, sink)]["output_slot_name"]
         )
+
+
+def test_loop_step() -> None:
+    step = STEP_KEYS["step_3"]
+    assert step.name == step.step_name == "step_3"
+    assert isinstance(step, LoopStep)
+    assert step.input_slots == {
+        "step_3_main_input": InputSlot(
+            "step_3_main_input",
+            "DUMMY_CONTAINER_MAIN_INPUT_FILE_PATHS",
+            validate_input_file_dummy,
+        )
+    }
+    assert step.output_slots == {"step_3_main_output": OutputSlot("step_3_main_output")}
+    assert isinstance(step.unlooped_node, BasicStep)
+    assert step.unlooped_node.name == step.name
+    assert step.unlooped_node.input_slots == step.input_slots
+    assert step.unlooped_node.output_slots == step.output_slots
+    assert step.edges == [
+        Edge(step.name, step.name, "step_3_main_output", "step_3_main_input")
+    ]
+
+
+def test_loop_update_implementation_graph(default_config: Config) -> None:
+    step = STEP_KEYS["step_3"]
```

Comment:
let me know whether this seems reasonable (appealing to the PipelineSchema to pull the instantiated Steps) and I will do it for all the other tests as well and save on defining all the parameters for each step.